subql_wika/src/mappings/eventHandlers.ts

```diff
@@ -0,0 +1,163 @@
+    */
+    logError(e) {
+        if (this.emails.isEnabled()) {
+            this.emails.sendError(e.toString()) ;
+        }
+        this.log.error(e);
+    }
+
+    /**
+    * Fetches a webpage and saves the metadata to Postgres and Elastic Search
+    * (when plugins are enabled)
+    * @param url - The url string
+    */
+    async updateMetadata(url) {
+        const metadata = await fetchMetadata(url) ;
+        if (metadata) {
+            if (this.postgres.isSyncEnabled()) {
+                this.postgres.newMetadataRecord(url, metadata) ;
+            } else {
+                this.log.warn('postgres is disabled')
+            }
+            if (this.elastic.isSyncEnabled()) {
+                await this.elastic.postUrl(url, metadata) ;
+            } else {
+                this.log.warn('elastic is disabled')
+            }
+        }
+    }
+
+    /**
+    * Processes a LikeEvent
+    * Updates the metadata in Postgres and Elastic Search
+    * Records the event in Postgres
+    * Creates or updates the LIKE relationship in Neo4J
+    * @param event - Polkadot API Event, containing user, url and numLikes in its data field
+    */
+    async handleLikeEvent(event) {
+        const eventData = event.event.data ;
+        const blockId = event.extrinsic.block.block.header.hash.toString() ;
+        const blockNum = event.extrinsic.block.block.header.number.toNumber();
+        const eventId = blockNum + '/' + event.idx ;
+        const user = eventData[0].toString() ;
+        const url = eventData[1].toHuman().toString() ;
+        const numLikes = Number(eventData[2]) ;
+        this.log.info('handleLikeEvent : '+eventId+' : '+user+' : '+url+' : '+numLikes) ;
+
+        // Update metadata
+        this.updateMetadata(url) ;
+
+        // Main record
+        if (this.postgres.isSyncEnabled()) {
+            this.postgres.newLikeEvent(eventId, blockNum, url, user, numLikes);
+        } else {
+            this.log.warn('postgres is disabled') ;
+        }
+
+        // Neo4J sync
+        if (this.neo4j.isSyncEnabled()) {
+            await this.neo4j.handleLikeEvent(user, url, numLikes) ;
+        } else {
+            this.log.warn('neo4j is disabled') ;
+        }
+    }
+
+    /**
+    * Processes a UrlRegisteredEvent
+    * Updates the metadata in Postgres and Elastic Search
+    * Records the event in Postgres
+    * Creates or updates the OWNS relationship in Neo4J
+    * @param event - Polkadot API Event, containing user and url in its data field
+    */
+    async handleUrlRegisteredEvent(event) {
+        const eventData = event.event.data ;
+        const user = eventData[0].toString() ;
+        const url = eventData[1].toHuman().toString() ;
+        const blockNum = eventData[2].toNumber();
+        const urlHash = crypto.createHash('md5').update(url).digest('hex');
+        const eventId = blockNum+'/'+urlHash ;
+        this.log.info('handleUrlRegisteredEvent : '+eventId+' : '+user+' : '+url) ;
```

Comment:
Just as a suggestion, here you could use js string templates, imho they offer better readability:

```typescript
this.log.info(`handleUrlRegisteredEvent : ${eventId} : ${user} : ${url}`);
```

Reply:
Cool, didn't know about it! Fixed.